medmodels/data_synthesis/synthesizer_model.pyi

```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+from typing import Union
+
+import sparse
+import torch
+from torch import nn
+
+from medmodels import MedRecord
+from medmodels.data_synthesis.synthesizer import Synthesizer
+
+class SynthesizerModel(nn.Module):
+    number_samples: int
+    device: torch.device
+
+    def __init__(
+        self,
+        synthesizer: Synthesizer,
```

Comment:
Abstract classes can have constructors, but they cannot be instantiated directly.

However, the idea is taking what is necessary from the Synthesizer and not saving it (mainly because it might be very big to save it twice). Let me know what you think though.